Lib9c/Action/Guild/BanGuildMember.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using Bencodex.Types;
+using Libplanet.Action;
+using Libplanet.Action.State;
+using Nekoyume.Extensions;
+using Nekoyume.Module.Guild;
+using Nekoyume.TypedAddress;
+
+namespace Nekoyume.Action.Guild
+{
+    [ActionType(TypeIdentifier)]
+    public class BanGuildMember : ActionBase
+    {
+        public const string TypeIdentifier = "ban_guild_member";
+
+        private const string TargetKey = "t";
+
+        public AgentAddress Target { get; private set; }
+
+        public override IValue PlainValue => Dictionary.Empty
+            .Add("type_id", TypeIdentifier)
+            .Add("values", Dictionary.Empty
+                .Add(TargetKey, Target.Bencoded));
+
+        public BanGuildMember() {}
+
+        public BanGuildMember(AgentAddress target)
+        {
+            Target = target;
+        }
+
+        public override void LoadPlainValue(IValue plainValue)
+        {
+            if (plainValue is not Dictionary root ||
+                !root.TryGetValue((Text)"values", out var rawValues) ||
+                rawValues is not Dictionary values ||
+                !values.TryGetValue((Text)TargetKey, out var rawTarget))
+            {
+                throw new InvalidCastException();
+            }
+
+            Target = new AgentAddress(rawTarget);
+        }
+
+        public override IWorld Execute(IActionContext context)
+        {
+            context.UseGas(1);
+            var world = context.PreviousState;
+
+            // NOTE: GuildMaster address and GuildAddress are the same with signer address.
+            var signer = context.GetAgentAddress();
+
+            if (world.GetJoinedGuild(signer) is not { } guildAddress)
+            {
+                throw new InvalidOperationException("The signer does not have a guild.");
+            }
+
+            if (!world.TryGetGuild(guildAddress, out var guild))
+            {
+                throw new InvalidOperationException("There is no such guild.");
+            }
+
+            if (guild.GuildMasterAddress != signer)
+            {
+                throw new InvalidOperationException("The signer is not a guild master.");
+            }
+
+            if (guild.GuildMasterAddress == Target)
+            {
+                throw new InvalidOperationException("The guild master cannot be banned.");
+            }
+
+            world = world.Ban(guildAddress, Target);
```

Comment:
It's best to run the Ban method last.
First do the guild reject, then leave the guild, then do the ban.
It shouldn't matter since it's done in one action anyway, but hmm...

Reply:
It was just an afterthought, a feature that was added later. The order doesn't seem to matter much. ✅

Reply:
I amended it in 48b20bd1237d35629b2dc4008c68a6594e0973e4